tools/toolchains/arm.py

```diff
@@ -285,6 +285,10 @@ def correct_scatter_shebang(self, sc_fileref, cur_dir_name=None):
         """
         with open(sc_fileref.path, "r") as input:
             lines = input.readlines()
+            if getattr(self.target, "default_lib", "std") == "small" and self.name != "ARMC6":                                           
+                for index, line in enumerate(lines):                
+                    if "__MICROLIB" in line:                
+                        lines[index]=lines[index].replace(lines[index],"#define __MICROLIB"+"\n"+lines[index])                                        
```

Comment:
It looks like this hunk of code should go inside the following else. Also a comment to say what is going on here would be really helpful when it comes to maintaining this code in future.
Also, it looks like you are trying to insert a new line into the 'lines' array; can you use 'lines.insert(index) instead?
Also, if we ever got to 'ARMC7' would this code still work? Ie is it really attempting to patch ARMC5 only, or anything not ARMC6.

Reply:
sorry it's draft version,  wrongly raised as a PR